### ISSUE TITLE
jq changes a long number value in output

### ISSUE THREAD

**Issue by Foxeronie** [jqlang/jq#3074]
**Describe the bug**
When I pipe a textfile output to jq, a long number gets manipulated/changed in the jq output.

**To Reproduce**
cat of a textfile, pipe into grep, all good. query_id is 5113603275375104668
```
root@node:~# cat /var/log/patroni/postgresql.json | grep 3932342
{"timestamp":"2024-03-19 08:48:10.687 CET","pid":3932342,"session_id":"65f9433a.3c00b6","line_num":1,"session_start":"2024-03-19 08:48:10 CET","vxid":"23/9450","txid":0,"error_severity":"FATAL","state_code":"57P01","message":"terminating background worker \"parallel worker\" due to administrator command","statement":"SELECT \"id\", \"patch\", \"lastContent\", \"content\", \"length\", \"authorship\", \"createdAt\", \"updatedAt\", \"noteId\" FROM \"Revisions\" AS \"Revision\" WHERE \"Revision\".\"noteId\" = '58e8bced-7c08-4484-a7fa-d744d80dcda0' ORDER BY \"Revision\".\"createdAt\" DESC;","backend_type":"parallel worker","leader_pid":3932337,"query_id":5113603275375104668}
```
Do the same and pipe it into jq, query_id is now 5113603275375105000
```
root@node:~# cat /var/log/patroni/postgresql.json | grep 3932342 | jq .
{
  "timestamp": "2024-03-19 08:48:10.687 CET",
  "pid": 3932342,
  "session_id": "65f9433a.3c00b6",
  "line_num": 1,
  "session_start": "2024-03-19 08:48:10 CET",
  "vxid": "23/9450",
  "txid": 0,
  "error_severity": "FATAL",
  "state_code": "57P01",
  "message": "terminating background worker \"parallel worker\" due to administrator command",
  "statement": "SELECT \"id\", \"patch\", \"lastContent\", \"content\", \"length\", \"authorship\", \"createdAt\", \"updatedAt\", \"noteId\" FROM \"Revisions\" AS \"Revision\" WHERE \"Revision\".\"noteId\" = '58e8bced-7c08-4484-a7fa-d744d80dcda0' ORDER BY \"Revision\".\"createdAt\" DESC;",
  "backend_type": "parallel worker",
  "leader_pid": 3932337,
  "query_id": 5113603275375105000
}
```

**Expected behavior**
The query_id field value should be the same.

**Environment (please complete the following information):**
Ubuntu 22.04
jq version 1.6

Best regards,
Patrick

**Comment by wader** [comment]
Hi, this was fixed in jq 1.7, see https://github.com/jqlang/jq/releases/tag/jq-1.7 about preserving precision for number literals
```sh
$ jq --version
jq-1.7.1

$ echo 5113603275375104668 | jq .
5113603275375104668

# fun fact, node truncates also
$ node -p 5113603275375104668
5113603275375105000
```
Note that jq will still truncate if you do any operation on the number
```sh
$ echo 5113603275375104668 | jq '.+0'
5113603275375105000
```

**Comment by Foxeronie** [comment]
Hi, thank you for the quick response!
